cpp/src/arrow/array/statistics_test.cc

```diff
@@ -96,6 +99,61 @@ TEST(ArrayStatisticsTest, TestEquality) {
   ASSERT_NE(statistics1, statistics2);
   statistics2.is_max_exact = true;
   ASSERT_EQ(statistics1, statistics2);
+
+  // Test different ArrayStatistics::ValueType
+  statistics1.max = static_cast<uint64_t>(29);
+  statistics1.max = static_cast<int64_t>(29);
+  ASSERT_NE(statistics1, statistics2);
+}
+class TestEqualityDoubleValue : public ::testing::Test {
```

Comment:
We want to use `TestArrayStatistics` prefix for all tests in this file because we mix multiple test source files for one test binary.
See also: https://github.com/apache/arrow/blob/23017e3c48c1033d48674415ee4453476dc61bf3/cpp/src/arrow/CMakeLists.txt#L1164-L1175

```suggestion
class TestArrayStatisticsEqualityDoubleValue : public ::testing::Test {
```